packages/core/src/providers/azure-ad.ts

```diff
@@ -121,20 +122,50 @@ export default function AzureAD<P extends AzureADProfile>(
      */
     profilePhotoSize?: 48 | 64 | 96 | 120 | 240 | 360 | 432 | 504 | 648
     /** @default "common" */
-    tenantId?: string
+    tenantId?: string,
   }
 ): OAuthConfig<P> {
   const { tenantId = "common", profilePhotoSize = 48, ...rest } = options
+  const groupedTenants = [
+    "common",
+    "organizations",
+    "consumers",
+    "consumers-preview",
+  ]
   rest.issuer ??= `https://login.microsoftonline.com/${tenantId}/v2.0`
   return {
     id: "azure-ad",
     name: "Azure Active Directory",
     type: "oidc",
     wellKnown: `${rest.issuer}/.well-known/openid-configuration?appid=${options.clientId}`,
+    token: {
+      async conform(response) {
+        // Microsoft being special and non-compliant #9635
+        response.url.replace("{tenantid}", tenantId);
```

Comment:
This doesn't do anything, does it? The Response `url` property is readonly and immutable

Reply:
You're probably right. I'm not a TS/JS dev but I should've picked something like this up.
I'd hold off on review until I say something.